### ISSUE TITLE
Queues: make `cmd` a 1-bit signal by removing `peek` functionality

### ISSUE THREAD

**Issue by polybeandip** [calyxir/calyx#2241]
None of our scheduling transactions require us to `peek` our queues. For this reason, we've decide to alter our shared interface to longer support it. 

**Updated Shared Interface**

- Input port `cmd`, a 1-bit integer. Selects the operation to perform:
  - `0`: `pop`.
  - `1`: `push`.
- Input port `value`, a 32-bit integer. The value to push.
- Input port `rank`, a 32-bit integer. The rank associated with `value` during a push. **[only for `binheap`]**
- Register `ans`, a 32-bit integer that is passed to the queue by reference.
If `pop` is selected, the queue writes the result to this register.
- Register `err`, a 1-bit integer that is passed to the queue by reference.
The queue raises this flag in case of overflow or underflow.

**Comment by KabirSamsi** [comment]
To further add to this, a specialization between Work-Conserving and Non-Work-Conserving Queues:

**WC**: Make `cmd` a 1-bit signal and remove the `peek` command – hence, it will flip between `push` and `pop`.
**NWC**: Make `cmd` a 1-bit signal and remove the `peek by value`, `peek by time` and `pop by time` commands - hence, we will flip between `push` and `pop by time`.

**Comment by anshumanmohan** [comment]
Hint for `fud` commands that give your resource estimates:
- https://github.com/cucapra/calyx-resource-eval/blob/main/sdn/scripts/sdn_four_ways_latency.sh
- https://github.com/cucapra/calyx-resource-eval/blob/main/sdn/scripts/sdn_four_ways_resources.sh

**Comment by calebmkim** [comment]
> What is the commit hash, and where can I find it?

The commit hash is a unique identifier created by git for each commit to a branch. You can think of the commit hash as specifying which version of the Calyx compiler you want to build and compile your designs with. Most of the recent commits should produce the same/similar code for a given design, so I don't think the commit matters too much, you can probably just pick a relatively recent one. 

If you click on "2,875 commits" button, you can see a full commit history with the hashes. 
<img width="921" alt="Screenshot 2024-08-01 at 1 45 59 PM" src="https://github.com/user-attachments/assets/fd680651-d509-4489-ba64-8deb192c2429">
<img width="1325" alt="Screenshot 2024-08-01 at 1 44 07 PM" src="https://github.com/user-attachments/assets/0a382e35-421a-47cc-9946-d0fbe009bffb">


> Also what are the params, as in what do they mean and what are they parameters to?

tl;dr you can probably just leave it empty. 

In Calyx, you can pass parameters to the compiler that gives specific information on how the compiler shold run. For example, there is a pass called `static-promotion` which converts dynamic Calyx code to static code. If you wanted to delete this pass you would do `-d static-promotion` to the Calyx compiler. Or, if you want to promote code, but not compact (i.e., automatically parallelize certain parts of the code during promotion) code then you would pass `-x static-promotion:compaction=false`. In the `params` this would correspond to "[static-promotion,compaction,false]". If you just want to run the default pipeline, then you can leave the field empty.
